contracts/sol6/KyberNetwork.sol

```diff
@@ -881,16 +895,11 @@ contract KyberNetwork is WithdrawableNoModifiers, Utils5, IKyberNetwork, Reentra
                 "invalid split bps"
             );
 
+//+ srcAmounts[i] <= srcAmount
+//*
             if (reservesData.isFeeAccountedFlags[i]) {
-                if (src == ETH_TOKEN_ADDRESS) {
-                    // reduce fee from srcAmount if fee paying.
-                    reservesData.srcAmounts[i] =
-                        ((srcAmount - networkFeeValue) * reservesData.splitsBps[i]) /
-                        BPS;
-                } else {
-                    reservesData.srcAmounts[i] = (srcAmount * reservesData.splitsBps[i]) / BPS;
-                    feesAccountedDestBps[i] = networkFeeBps;
-                }
+                reservesData.srcAmounts[i] = srcAmountAfterFee * reservesData.splitsBps[i] / BPS;
```

Comment:
Can you explain the reason why move the  calculation outside of the if block

Reply:
srcAmountAfterFee is calculated for src = ETH_TOKEN_ADDRESS
so when ETH_TOKEN_ADDRESS we always subtract networkFeeValue. so just did one subtraction.
if src != ETH_TOKEN_ADDRESS it will just be srcAmount.
So should keep same behavior

Reply:
can resolve conversation?